src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1380,13 +1446,28 @@ public sealed class ValidateSetAttribute : ValidateEnumeratedArgumentsAttribute
         public bool IgnoreCase { get; set; } = true;
 
         /// <summary>
-        /// Gets the values in the set
+        /// Gets the valid values in the set.
         /// </summary>
         public IList<string> ValidValues
         {
             get
             {
-                return _validValues;
+                if (validValuesGenerator == null)
+                {
+                    return _validValues;
+                }
+
+                var validValuesLocal = validValuesGenerator.GetValidValues();
```

Comment:
It seems related #3769 - completers silently falls back to paths.
I tested a generator with `throw "Test throw!"` - parameter binder handles any exceptions:
```powershell
Get-TestValidateSetPS4 : Cannot validate argument on parameter 'Param1'. Test throw!
At line:1 char:32
+ Get-TestValidateSetPS4 -param1 fdfd
+                                ~~~~
    + CategoryInfo          : InvalidData: (:) [Get-TestValidateSetPS4], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationError,Get-TestValidateSetPS4
```
If the generator throws good formatted exception users get it in InnerException.
So it seems redundant to add extra level to catch the exception.